### DISCUSSION TITLE
Select on join and condition

### DISCUSSION THREAD

**Discussion by ramadom** [mybatis/mybatis-dynamic-sql#857] [Q&A] [answered]
Does Mybatis Dynamic support a sub select inside and() method as a join() condition?
```
SELECT ...
FROM table1 t1
JOIN table2 t2 ON t2.id = t1.id AND t2.value = (SELECT max(column) FROM table3)
```
How it should be defined in this case?
```
select(...)
.from(table1, "t1")
.join(table2, "t2").on(t2.id, equalTo(t1.id), and(t2.value, isEqualTo(select(max(column)).from(table3)))
```
Same and() condition seems allowed with where() clause.

**Comment by jeffgbutler** [comment, marked as answer]
This is supported in the 2.0.0 version. It's not released yet, but it is in the SNAPSHOT version currently.

**Reply by ramadom** [reply]
Will check v2.0.0 and wait for release. Thanks for the great tool!
